src/test/regress/sql/pg13.sql

```diff
@@ -94,4 +94,14 @@ SELECT create_distributed_table('test_table', 'a');
 -- we currently don't support this
 CREATE INDEX test_table_index ON test_table USING gist (b tsvector_ops(siglen = 100));
 
+-- testing WAL
+CREATE TABLE test_wal(a int, b int);
+-- test WAL without ANALYZE, this should raise an error
+EXPLAIN (WAL) INSERT INTO test_wal VALUES(1,11);
+-- test WAL working properly
+EXPLAIN (ANALYZE TRUE, WAL TRUE, COSTS FALSE, SUMMARY FALSE, BUFFERS FALSE, TIMING FALSE) INSERT INTO test_wal VALUES(1,11);
+SELECT create_distributed_table('test_wal', 'a');
+EXPLAIN (ANALYZE TRUE, WAL TRUE, COSTS FALSE, SUMMARY FALSE, BUFFERS FALSE, TIMING FALSE) INSERT INTO test_wal VALUES(2,22);
+EXPLAIN (ANALYZE TRUE, WAL TRUE, COSTS FALSE, SUMMARY FALSE, BUFFERS FALSE, TIMING FALSE) DELETE FROM test_wal WHERE a=2;
```

Comment:
Can you add a query where the resulting plan contains a subplan (e.g. a modifying CTE).

After reading your updated code it seems WAL is closer to BUFFERS than to SUMMARY. Since there's some extra code for BUFFERS to handle subplan, I can imagine you will need to do something similar to make them work.